### PR TITLE
Update tests structure.

### PR DESCRIPTION
**Pull request by downdawn**
Hi, @wu-clan 

Is such a test more reasonable?

The development environment itself can also be considered as the test environment, depending on the requirements of various specifications. 

Or add a database dedicated to testing.
[https://fastapi.tiangolo.com/advanced/testing-database/](https://fastapi.tiangolo.com/advanced/testing-database/)

### CODE REVIEW

File: backend/app/tests/api_v1/test_auth.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        'username': settings.TEST_USERNAME,
+        'password': settings.TEST_USER_PASSWORD,
+    }
+    response = client.post(f'{settings.API_V1_STR}/auth/login', json=data)
+    assert response.status_code == 200
+    assert response.json()['data']['access_token_type'] == 'Bearer'
+
+
+def test_logout(client: TestClient, token_headers: dict[str, str]) -> None:
+    response = client.post(f'{settings.API_V1_STR}/auth/logout', headers=token_headers)
```
Comment (wu-clan) [inline backend/app/tests/api_v1/test_auth.py:19]: Will this test pass?

File: backend/app/tests/utils/get_headers.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+def get_token_headers(client: TestClient, username: str, password: str) -> Dict[str, str]:
+    data = {
+        'username': username,
+        'password': password,
+    }
+    response = client.post(f'{settings.API_V1_STR}/auth/login', json=data)
+    token_type = response.json()['data']['access_token_type']
+    access_token = response.json()['data']['access_token']
+    headers = {'Authorization': f'{token_type} {access_token}'}
+    return headers
```
Comment (wu-clan) [inline backend/app/tests/utils/get_headers.py:19]: This will store n user tokens in redis?

File: backend/app/.env.example [hunk 3]
```diff
@@ -18,3 +18,6 @@ APS_REDIS_DATABASE=1
 # Token
 TOKEN_SECRET_KEY='1VkVF75nsNABBjK_7-qz7GtzNy3AMvktc9TCPwKczCk'
 TOKEN_WHITE_LIST=[1]
+# Test
+TEST_USERNAME='test'
+TEST_USER_PASSWORD='123456'
```
Comment (wu-clan) [inline backend/app/.env.example:23]: Will test-exclusive data be included only in the test?

**Comment by wu-clan** [comment]
This is a repository I created last week [fastapi_best_architecture_tests](https://github.com/fastapi-practices/fastapi_best_architecture_tests)

My initial idea was to add all the tests based on my other project ([automated_api_pytest](https://github.com/wu-clan/automated_api_pytest)), but no one should be willing to spend the learning cost for that

File: backend/app/tests/api_v1/test_auth.py (hunk 1 above)
Comment (downdawn) [reply backend/app/tests/api_v1/test_auth.py:19]: Yes

File: backend/app/tests/utils/get_headers.py (hunk 2 above)
Comment (downdawn) [reply backend/app/tests/utils/get_headers.py:19]: Not if the test user is not in the multipoint login whitelist.

File: backend/app/.env.example (hunk 3 above)
Comment (downdawn) [reply backend/app/.env.example:23]: It can also be used to initialize data

**Comment by downdawn** [comment]
> This specification works
> 
> In addition, we need to isolate test data, not just database related content

No problem, can set up a separate test database by referring to the official example

**Comment by downdawn** [comment]
> This is a repository I created last week [fastapi_best_architecture_tests](https://github.com/fastapi-practices/fastapi_best_architecture_tests)
> 
> My initial idea was to add all the tests based on my other project ([automated_api_pytest](https://github.com/wu-clan/automated_api_pytest)), I haven't implemented it yet, because adding tests based on automated_api_pytest, I still need to modify it a lot in order to provide the user with the easiest way to execute it, it takes time, but, I don't believe anyone spends a lot of learning costs for testing
> 
> So, my idea is to add only part of the test example to this code and leave the rest to the users themselves

Agree, expect

File: backend/app/tests/utils/get_headers.py (hunk 2 above)
Comment (wu-clan) [reply backend/app/tests/utils/get_headers.py:19]: Yes.

File: backend/app/.env.example (hunk 3 above)
Comment (wu-clan) [reply backend/app/.env.example:23]: Sounds good.

But I don't think the test data should be in the environment, just in the config, as this is only for testing purposes

**Comment by wu-clan** [comment]
> No problem, can set up a separate test database by referring to the official example

Can you submit the code?

**Comment by wu-clan** [comment]
In addition, need to update init_test_data.py and README

**Comment by downdawn** [comment]
please review

**Comment by wu-clan** [comment]
This is so cool.

**Comment by wu-clan** [comment]
LGTM.

**Review by wu-clan — APPROVED** [review]

**Pull request merged**